pkg/mover/intermediate.go

```diff
@@ -119,9 +129,113 @@ func tarImages(imageChanges []*internal.ImageChange, logger Logger) (string, err
 		logger.Printf("Processing image %s\n", change.ImageReference.Name())
 	}
 
-	logger.Printf("Writing all %d images...\n", len(refToImage))
+	logger.Printf("Writing %d images...\n", len(refToImage))
 	if err := tarball.MultiRefWrite(refToImage, imagesFile); err != nil {
 		return "", err
 	}
 	return imagesFile.Name(), nil
 }
+
+// IsIntermediateBundle returns tue only if VerifyIntermediateBundle finds no errors
+func IsIntermediateBundle(bundlePath string) bool {
+	return VerifyIntermediateBundle(bundlePath) == nil
+}
+
+type fileValidations struct {
+	filename, format string
+	validate         func(io.Reader) error
+}
+
+// VerifyIntermediateBundle returns true if the path points to an uncompressed
+// tarball with:
+//  A hints.yaml YAML file
+//  A manifest.json for the images
+//  A directory container an unpacked chart directory with valid YAMLs Chart.yaml & values.yaml
+func VerifyIntermediateBundle(bundlePath string) error {
+	validations := []fileValidations{
+		{filename: "hints.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/Chart.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/values.yaml", format: "YAML", validate: validateYAML},
```

Comment:
Would it make sense to validate this with the Helm library instead similarly to what we do when we load it?

Reply:
or we simplify validation further and delay full validation till actual usage.

Reply:
simplied validation